pkg/controller/launch/pool_manager.go

```diff
@@ -113,13 +135,31 @@ func (cpm *ConcretePoolManager) SetStatus(status *PoolStatus) error {
 		return err
 	}
 
-	for i, curInfo := range copy.Status.NodePools {
-		if curInfo.Name == newInfo.Name {
-			if curInfo == newInfo {
+	// Add new pools in the spec to the status
+	for _, specPool := range copy.Spec.NodePools {
+		// Find the pool
+		if npi, ok := nodePoolInfoGet(copy.Status.NodePools, specPool.Name); ok {
+			// is there a need to update?
+			if copy.Status.NodePools[npi] == newInfo {
 				return nil
 			}
+			copy.Status.NodePools[npi] = newInfo
+			_, err = cpm.Clients.Kubernikus.Kubernikus().Klusters(copy.Namespace).Update(copy)
+			return err
```

Comment:
returning here would skip the rest of the loop, no? This would be solved by doing only one update in the end and returning that error.

Reply:
yes you are right i've leaned too much on the old code and have to revisit the returns.